src/lazy/encoder/binary/v1_1/flex_uint.rs

```diff
@@ -129,147 +98,30 @@ impl FlexUInt {
-        //      b4       b5       b6       b7
-        //   MMMMMMMM MMMMMMMM MMMMMMMM MMMMMMMM
-        //      b8       b9
-        //   MMMMMMMM XXXXXXMM
-        //
-        // P = Prefix bit
-        // M = Magnitude bit
-        // X = An 'extra' bit; if `support_sign_extension` is true, these are sign bits.
-
-        // We've already processed the first byte, and we've looked at the lowest two bits of
-        // the second byte. Isolate the highest six bits of the second byte (b1) which represent
-        // the lowest six bits of the magnitude.
-        let magnitude_low_six = second_byte >> 2;
-        // Load the remaining 8 bytes into a u64 that we can easily shift/mask.
-        let remaining_data = &input[2..10];
-        // We know the slice is 8 bytes long, so we can `unwrap()` the conversion to [u8; 8]
-        let remaining_magnitude = u64::from_le_bytes(remaining_data.try_into().unwrap());
-
-        let sign_extension_bits = (remaining_magnitude & (0b111111 << 58)) >> 58;
-        if support_sign_extension {
-            // Something downstream intends to use this as a signed value; we need to make sure
-            // that bits 65-70 match bit 64. `remaining_magnitude` is storing 58 bits of data,
-            // so bit 64 of the value (bit index=63) is bit 58 (bit index=57) in `remaining_magnitude`.
-            let high_bit_is_set = remaining_magnitude & (1 << 57) != 0;
-            if (high_bit_is_set && sign_extension_bits != 0b111111)
-                || (!high_bit_is_set && sign_extension_bits != 0)
-            {
-                // If the sign extension bits don't agree with the top bit, this value required
-                // more than 64 bits to encode.
-                return IonResult::decoding_error(
-                    "found a 10-byte FlexInt too large to fit in a i64",
-                );
-            }
-        } else {
-            // This is an unsigned value; if any of the highest six bits are set, then this
-            // value is beyond the magnitude we can store in a u64.
-            if sign_extension_bits != 0 {
-                return IonResult::decoding_error(
-                    "found a 10-byte FlexUInt too large to fit in a u64",
-                );
-            }
+        if num_encoded_bytes > input.len() {
+            return incomplete();
         }
-
-        // Shift the magnitude from the last 8 bytes over and combine it with the six bits we
-        // carried over from the second byte.
-        let value = (remaining_magnitude << 6) | magnitude_low_six as u64;
-        let flex_uint = FlexUInt::new(10, value);
-        Ok(flex_uint)
+        let mut buffer = [0u8; size_of::<u128>()];
+        buffer[..num_encoded_bytes].copy_from_slice(&input[..num_encoded_bytes]);
+        let big_value = u128::from_le_bytes(buffer).wrapping_shr(num_encoded_bytes as u32);
+        let value = big_value as u64;
+        Ok(FlexUInt::new(num_encoded_bytes, value))
```

Comment:
🪧 In the new version, the serialized form fits entirely in a `u128` so we can use standard bitshifting to eliminate the continuation flag bits.